File: app/src/main/java/com/asmat/rolando/bakingapp/fragments/StepsFragment.java
package com.asmat.rolando.bakingapp.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.asmat.rolando.bakingapp.R;
import com.asmat.rolando.bakingapp.adapters.StepsAdapter;
import com.asmat.rolando.bakingapp.models.Step;

import java.util.ArrayList;

/**
 * Created by rolandoasmat on 8/15/17.
 */

public class StepsFragment extends Fragment {
    private Step[] mItems;
    private static final String ARG_STEPS = "arg_steps";

    @Override
    public void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if(getArguments() != null) {
            mItems = (Step[]) getArguments().getParcelableArray(ARG_STEPS);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater,  ViewGroup container,  Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_steps, container, false);
        Context context = view.getContext();
        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.list);
        recyclerView.setLayoutManager((new LinearLayoutManager(context)));
        ArrayList<Step> list = new ArrayList();
        for(Step step: mItems) {
            list.add(step);
        }
        recyclerView.setAdapter(new StepsAdapter(list));
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if(!(context instanceof StepsFragment.OnStepsFragmentInteractionListener)) {
            throw new RuntimeException(context.toString() + " must implement OnStepsFragmentInteractionListener");
        }
    }

    public interface OnStepsFragmentInteractionListener {
        void onBeginRecipe(View view);
    }

    public final static StepsFragment newInstance(Step[] steps) {
        StepsFragment fragment = new StepsFragment();
        Bundle args = new Bundle();
        args.putParcelableArray(ARG_STEPS, steps);
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/asmat/rolando/bakingapp/fragments/StepsFragment.kt
package com.asmat.rolando.bakingapp.fragments

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.asmat.rolando.bakingapp.R
import com.asmat.rolando.bakingapp.adapters.StepsAdapter
import com.asmat.rolando.bakingapp.models.Step

/**
 * Created by rolandoasmat on 7/12/17.
 */

class StepsFragment : Fragment() {
    private var mItems: Array<Step>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mItems = arguments.getParcelableArray(ARG_STEPS) as Array<Step>
        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_steps, container, false)

        val context = view.context
        val recyclerView = view.findViewById(R.id.list) as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(context)
        val list = ArrayList<Step>()
        for(step in mItems!!) {
            list.add(step)
        }
        recyclerView.adapter = StepsAdapter(list)
        return view
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context !is OnStepsFragmentInteractionListener) {
            throw RuntimeException(context!!.toString() + " must implement OnStepsFragmentInteractionListener")
        }
    }

    interface OnStepsFragmentInteractionListener {
        fun onBeginRecipe(view:View)
    }

    companion object {
        private val ARG_STEPS= "arg_steps"

        fun newInstance(steps: Array<Step>): StepsFragment {
            val fragment = StepsFragment()
            val args = Bundle()
            args.putParcelableArray(ARG_STEPS, steps)
            fragment.arguments = args
            return fragment
        }
    }
}
